ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -328,10 +336,34 @@ public BlobIdType getType() {
   /**
    * @return {@code true} if the blob that this id represents is encrypted. {@code false} otherwise
    */
-  public boolean isEncrypted() {
+  private boolean isEncrypted() {
     return isEncrypted;
   }
 
+  /**
+   * Check if a blob is encrypted based on its encrypted bit in blobId.
+   * For BLOB_ID_V3, original blobIdString is needed to check encrypted bit.
+   * @return {@code true} if the blob that this id represents is encrypted. {@code false} otherwise
+   */
+  public static boolean isEncrypted(BlobId blobId, String blobIdString) {
```

Comment:
Just noting what we discussed in person. The arguments to this method are redundant and leaves room for errors. Callers to this method already seem to have the blob id string, so let us see if we can change this to `public static boolean isEncrypted(String blobIdStr);`

Reply:
My initial try was add isEncrypted to `void submitGetBlobOperation(BlobId blobId, GetBlobOptionsInternal options, Callback<GetBlobResultInternal> callback)` as another argument, but it didn't look good and too many changes involved. 

Now, I am trying to add isEncrypted as a filed in `GetBlobOptionsInternal`. This makes the API looks more nature and easy to extend, for future: compression bit can be added.

Reply:
Actually, `isEncrypted` is a bit out of place in `GetBlobOptionsInternal` as it is not a parameter to the Get Operation. Is it possible to change the submitGetBlobOperation to take `blobIdStr` instead of `blobId`?

Reply:
That will introduce `isEncrypted` to `GetBlobInfoOperation(routerConfig, routerMetrics, clusterMap, responseHandler, blobId, options, callback,
              routerCallback, kms, cryptoService, cryptoJobHandler, time);` and `GetBlobOperation(routerConfig, routerMetrics, clusterMap, responseHandler, blobId, options, callback,
              routerCallback, blobIdFactory, kms, cryptoService, cryptoJobHandler, time)`

I am OK with this. I was thinking one `option` is more concise than `option, isEncrypted` and less changes in other places, like exception handling.

Will send out PR soon.